spec/ktl/continous_reassigner_spec.rb

```diff
@@ -0,0 +1,163 @@
+      end
+
+      let :znodes do
+        {}
+      end
+
+      before do
+        allow(zk_client).to receive(:reassign_partitions) do |r|
+          reassignments << JSON.parse(r).fetch('partitions')
+        end
+        allow(zk_client).to receive(:create_znode) do |path, data|
+          p [:create, :path, path]
+          znodes[path] = data
+        end
+        allow(zk_client).to receive(:delete_znode) do |path|
+          p [:delete, :path, path]
+          znodes.delete(path)
+        end
+        allow(zk_client).to receive(:get_children) do |path|
+          matching = znodes.select { |k, v| k.start_with?(path) }
+          matching = matching.keys.map { |s| s.split('/').last }
+          p [:children, path, matching]
+          scala_list(matching)
+        end
+        allow(zk_client).to receive(:exists?) do |path|
+          znodes.key?(path)
+        end
+        allow(zk_client).to receive(:read_data) do |path|
+          p [:read_data, path]
+          [znodes[path]]
+        end
+        allow(zk_client).to receive(:watch_data)
+        allow(zk_client).to receive(:unsubscribe_data)
+      end
+
+      def reassign
+        latch = JavaConcurrent::CountDownLatch.new(1)
+        t = Thread.new { latch.count_down; reassigner.execute(reassignment) }
+        latch.await
+        if block_given?
+          yield t
+        else
+          reassigner.handle_data_deleted(Kafka::Utils::ZkUtils.reassign_partitions_path)
+        end
+        t.join
+        expect(t.status).to eq(false)
+      end
+
+      it 'reassigns partitions' do
+        reassign
+        expect(zk_client).to have_received(:reassign_partitions).with(json)
+      end
+
+      it 'watches the reassignment path for data changes' do
+        reassign
+        expect(zk_client).to have_received(:watch_data).with(Kafka::Utils::ZkUtils.reassign_partitions_path, reassigner)
+      end
+
+      it 'removes the watcher when the reassignment is done' do
+        reassign
+        expect(zk_client).to have_received(:unsubscribe_data).with(Kafka::Utils::ZkUtils.reassign_partitions_path, reassigner)
+      end
+
+      context 'when the reassignment is greater than 1MB' do
+        let :reassignment do
+          r = Scala::Collection::Map.empty
+          25_000.times.each do |partition|
+            topic_partition = Kafka::TopicAndPartition.new('topic1', partition)
+            replicas = scala_int_list([0, 1, 2])
+            r += Scala::Tuple.new(topic_partition, replicas)
+          end
+          r
+        end
+
+        it 'splits each reassignment' do
+          reassign do
+            sleep 0.1 until znodes.any? { |k, _| k == '/ktl/reassign' }
+            overflow = znodes.select { |k, v| k.start_with?('/ktl/overflow') }
+            expect(overflow).to_not be_empty
+            while znodes.any? { |k, _| k == '/ktl/reassign' }
+              reassigner.handle_data_deleted(Kafka::Utils::ZkUtils.reassign_partitions_path)
+            end
```

Comment:
This loop is repeated in a few tests, and I don't really understand what it does, maybe you can move it to a method and call it something that makes it easier to understand why it's necessary.